servers/src/test/java/tachyon/worker/block/evictor/LRFUEvictorTest.java

```diff
@@ -0,0 +1,297 @@
+    TachyonConf conf = new TachyonConf();
+    conf.set(Constants.WORKER_EVICT_STRATEGY_CLASS, LRFUEvictor.class.getName());
+    mStepFactor = conf.getDouble(Constants.WORKER_EVICT_STRATEGY_LRFU_STEP_FACTOR, 0.25);
+    mAttenuationFactor =
+        conf.getDouble(Constants.WORKER_EVICT_STRATEGY_LRFU_ATTENUATION_FACTOR, 2.0);
+    mEvictor = Evictor.Factory.createEvictor(conf, mManagerView);
+  }
+
+  private void cache(long userId, long blockId, long bytes, int tierLevel, int dirIdx)
+      throws Exception {
+    StorageDir dir = mMetaManager.getTiers().get(tierLevel).getDir(dirIdx);
+    EvictorTestUtils.cache(userId, blockId, bytes, dir, mMetaManager, mEvictor);
+  }
+
+  // access the block to update evictor
+  private void access(long blockId) {
+    ((BlockStoreEventListener) mEvictor).onAccessBlock(USER_ID, blockId);
+  }
+
+  /**
+   * Sort all blocks in ascending order of CRF
+   * 
+   * @return the sorted CRF of all blocks
+   */
+  private List<Map.Entry<Long, Double>> getSortedCRF(Map<Long, Double> crfMap) {
+    List<Map.Entry<Long, Double>> sortedCRF =
+        new ArrayList<Map.Entry<Long, Double>>(crfMap.entrySet());
+    Collections.sort(sortedCRF, new Comparator<Map.Entry<Long, Double>>() {
+      public int compare(Entry<Long, Double> o1, Entry<Long, Double> o2) {
+        double res = o1.getValue() - o2.getValue();
+        if (res < 0) {
+          return -1;
+        } else if (res > 0) {
+          return 1;
+        } else {
+          return 0;
+        }
+      }
+    });
+    return sortedCRF;
+  }
+
+  @Test
+  public void evictInBottomTierTest() throws Exception {
+    int bottomTierLevel = EvictorTestUtils.TIER_LEVEL[EvictorTestUtils.TIER_LEVEL.length - 1];
+    Map<Long, Double> blockIdToCRF = new HashMap<Long, Double>();
+    // capacity increases with index
+    long[] bottomTierDirCapacity = EvictorTestUtils.TIER_CAPACITY[bottomTierLevel];
+    int nDir = bottomTierDirCapacity.length;
+    // fill in dirs from larger to smaller capacity with blockId equal to BLOCK_ID plus dir index
+    for (int i = 0; i < nDir; i ++) {
+      cache(USER_ID, BLOCK_ID + i, bottomTierDirCapacity[i], bottomTierLevel, i);
+      // update CRF of blocks when blocks are committed
+      blockIdToCRF.put(BLOCK_ID + i, Math
+          .pow(1.0 / mAttenuationFactor, mStepFactor * (nDir - 1 - i)));
+    }
+    // access blocks in the order: 10, 10, 11, 10, 11, 12. Update CRF of all blocks
+    // during each access
+    for (int i = 0; i < nDir; i ++) {
+      for (int j = 0; j <= i; j ++) {
+        access(BLOCK_ID + j);
+        for (int k = 0; k < nDir; k ++) {
+          if (k == j) {
+            blockIdToCRF.put(BLOCK_ID + k,
+                blockIdToCRF.get(BLOCK_ID + k) * Math.pow(1.0 / mAttenuationFactor, mStepFactor)
+                    + 1.0);
+          } else {
+            blockIdToCRF.put(BLOCK_ID + k,
+                blockIdToCRF.get(BLOCK_ID + k) * Math.pow(1.0 / mAttenuationFactor, mStepFactor));
```

Comment:
For each access, logic time count will increment. If `k != j`, CRF will attenuate a bit. if 'k == j' old CRF will attenuate too, but the attenuated CRF will add another 1.0 for the current access.